docs/reference/style-guides/node-style-guide.md

```diff
@@ -2,8 +2,186 @@
 title: Node Style Guide
 ---
 
-Current as of `July 18, 2022`
+Current as of `January 18, 2023`
 
-## Introduction
+## Code Standards
 
-Building...
+These code standards apply to new packages in the FxA monorepo, such as:
+
+- `packages/fxa-graphql-api`
+- `packages/fxa-event-broker`
+
+Most of the standards in the code is enforced with [Prettier][2] and [ESLint][1].
+
+### Style
+
+We use the [Google TypeScript Style Guide](https://google.github.io/styleguide/tsguide.html#source-organization) for our base code style.
+
+Code should also follow the [Do's and Don'ts of TypeScript](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html).
+
+### Structure
+
+#### Organize by Feature
+
+(This is present in the Google TypeScript Style Guide, but bears repeating)
+
+Code should be organized by feature, not by type.
+
+Don't:
+
+```
+app/
+├── controllers/
+│   └── auth.ts
+├── models/
+│   └── auth.ts
+├── validators/
+│   └── auth.ts
+```
+
+Do:
+
+```
+app/
+├── auth/
+│   ├── auth.controller.ts
+│   ├── auth.model.ts
+│   └── auth.validator.ts
+```
+
+If a feature has many DTOs, models, etc., a hybrid approach may be used, organized by feature first:
+
+```
+app/
+├── auth/
+│   ├── dto/
+│   ├── model/
+│   |   └── auth.model.ts
+│   ├── auth.controller.ts
+│   └── auth.validator.ts
+```
+
+Tests should be organized alongside the code being tested, with the `.spec.ts` suffix:
+
+```
+app/
+├── auth/
+│   ├── auth.controller.ts
+│   ├── auth.controller.spec.ts
```

Comment:
Can we add an example of integration test spec?

Reply:
Yep, did we say this would be `.int.ts` or something? I forget what we had discussed on this point for the convention here.